src/modm/architecture/interface/i2c.cpp

```diff
@@ -25,21 +25,21 @@ operator << (modm::IOStream& s, const modm::I2c::Operation op)
 }
 
 modm::IOStream&
-operator << (modm::IOStream& s, const modm::I2c::OperationAfterStart op)
+modm::operator << (modm::IOStream& s, const modm::I2c::OperationAfterStart op)
```

Comment:
There is another operator in line 15 in this file which was not changed, it should not compile.

Reply:
Well spotted!

>  it should not compile

But it did in the CI. So there is a test missing.

Shall we move [`MemoryWriter`](https://github.com/modm-io/modm/blob/develop/test/modm/io/io_stream_test.cpp#L29) which implements `modm::IODevice` to `mock` and use it to test these stream operators?

Reply:
> Shall we move MemoryWriter which implements modm::IODevice to mock and use it to test these stream operators?

Yes please.

Reply:
Isn't `modm::IODevice` an interface? Does [IODevice.hpp](https://github.com/modm-io/modm/blob/develop/src/modm/io/iodevice.hpp) belong to [`interface`](https://github.com/modm-io/modm/tree/develop/src/modm/architecture/interface)?

And [`io_device.hpp`](https://github.com/modm-io/modm/blob/develop/test/modm/mock/io_device.hpp) with `FakeIODevice` probably needs a new name?

Reply:
I always though of architecture/interface as software abstractions of hardware. IODevice is pure software though, so it doesn't really fit in there (even though it had different implementations per platform for efficiency).

Reply:
> Yes please.

Done. I'll leave the `io_device.hpp` and `FakeIODevice` for now.